ui/components/app/asset-list/asset-list.js

```diff
@@ -126,7 +126,41 @@ const AssetList = ({ onClickAsset }) => {
   const defaultSwapsToken = useSelector(getSwapsDefaultToken);
   ///: END:ONLY_INCLUDE_IF
 
+  const trackDetectedTokens = useCallback(() => {
```

Comment:
Is this telemetry related to turning autodetection on, or a separate improvement?

Reply:
it's a separate one, basically we want to:
1. Track the number of tokens that are auto imported but hidden by the user by Network  
2. Track the number of new token instances imported by Network

Reply:
I'd be okay not doing this in this PR, since we have a separate telemery task.

Would this fire an HTTP request every time I open metamask?  I haven't worked with metrics before, but I'm not sure this is the best place to track this.  

I'm thinking it would be better to do in the core TokensController.   That way we only fire the event when the number of tokens actually changes, instead of during rendering a component.  The controllers appear to have access to `trackMetaMetricsEvent`, and doing it there would also make it work for mobile and extension.

Reply:
you mean on the `TokenDetectionController` ?  i just added a PR on the core repo [here](https://github.com/MetaMask/core/pull/4162) let me know if it corresponds to your suggestion

Reply:
I'll look at that change.  I'd like to remove the remaining metics changes from this PR so i can do them in my task.  They seem separate from changing the autodetection setting.